sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c

```diff
@@ -125,77 +187,65 @@ bool test_main(void) {
   if (rstmgr_reset_info == kDifRstmgrResetInfoPor) {
     LOG_INFO("POR reset");
 
-    LOG_INFO("Scrambling...");
-    sram_ctrl_testutils_scramble(&ret_sram);
-
-    // Write data to retention SRAM and readback (to test basic functionality.)
-    retention_sram_check((check_config_t){.do_write = true, .is_equal = true});
-
-    aon_timer_testutils_wakeup_config(&aon_timer, 100);
-    // Enable all the AON interrupts used in this test.
-    rv_plic_testutils_irq_range_enable(&rv_plic, kTopEarlgreyPlicTargetIbex0,
-                                       kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
-                                       kTopEarlgreyPlicIrqIdPwrmgrAonWakeup);
-    CHECK_DIF_OK(dif_pwrmgr_irq_set_enabled(&pwrmgr, 0, kDifToggleEnabled));
-
-    // Normal sleep.
-    pwrmgr_testutils_enable_low_power(
-        &pwrmgr, /*wakeups=*/kDifPwrmgrWakeupRequestSourceFive,
-        /*domain_config=*/kDifPwrmgrDomainOptionCoreClockInLowPower |
-            kDifPwrmgrDomainOptionUsbClockInActivePower |
-            kDifPwrmgrDomainOptionMainPowerInLowPower);
-    // Enter low power mode.
-    LOG_INFO("Issue WFI to enter normal sleep");
-    wait_for_interrupt();
-    LOG_INFO("Awaked");
-    // Check that the previously written retention SRAM data can still
-    // be read successfully.
-    retention_sram_check((check_config_t){.do_write = false, .is_equal = true});
-
-    // Prepare rstmgr for a reset.
-    rstmgr_testutils_pre_reset(&rstmgr);
-    aon_timer_testutils_wakeup_config(&aon_timer, 20);
-    // Deep sleep.
-    pwrmgr_testutils_enable_low_power(&pwrmgr,
-                                      kDifPwrmgrWakeupRequestSourceFive, 0);
+    test_ret_sram_in_normal_sleep();
 
-    // Enter low power mode.
-    LOG_INFO("Issue WFI to enter deep sleep");
-    wait_for_interrupt();
+    enter_deep_sleep();
```

Comment:
whenever we enter deep sleep or issue a reset request, should we add a `CHECK(false, "should have reset before this line")` or similar? just to ensure that the test cannot accidentally run through to the end and finish before it should have actually finished?

Reply:
Good idea. Added. Thanks